### PR TITLE
updating CODEOWNERS as per issue #146

### LINKED ISSUE #146: Baseline MAINTAINERS, CODEOWNERS, and external collaborator permissions
Follow https://github.com/opensearch-project/.github/issues/125 to baseline MAINTAINERS, CODEOWNERS, and external collaborator permissions.

Close this issue when:

1. [MAINTAINERS.md](MAINTAINERS.md) has the correct list of project maintainers.
2. [CODEOWNERS](CODEOWNERS) exists and has the correct list of aliases.
3. Repo permissions only contain individual aliases as collaborators with maintain rights, admin, and triage teams. 
4. All other teams are removed from repo permissions.

If this repo's permissions was already baselined, please confirm the above when closing this issue.

### PR DESCRIPTION
**Pull request by macohen**
### Description
Addresses new standards to make maintainers clearer/more transparent 

### Issues Resolved
Closes #146 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by noCharger — APPROVED** [review]

**Comment by noCharger** [comment]
Do we need to change the merge permission as well? Right now approval from anyone of https://github.com/orgs/opensearch-project/teams/search-relevance-dashboards-team is enough for PR merge.

**Comment by macohen** [comment]
> Do we need to change the merge permission as well? Right now approval from anyone of https://github.com/orgs/opensearch-project/teams/search-relevance-dashboards-team is enough for PR merge.

Yes. That's the next step after this (step 6 in https://github.com/opensearch-project/.github/issues/125). Before merging I need to just say hi to @dblock and ask for an assist here...

**Review by noCharger — APPROVED** [review]

**Pull request merged**
